jerry-core/jerry.c

```diff
@@ -119,526 +124,565 @@ jerry_make_api_available (void)
-} /* jerry_create_string_value */
+  if (parse_status != NULL)
+  {
+    return ecma_make_error_obj_value (parse_status);
+  }
 
-/**
- * Convert completion of 'eval' to API value and completion code
- *
- * Note:
- *      The value returned in 'retval_p' should be freed with jerry_release_value
- *
- * @return completion code
- */
-static inline jerry_completion_code_t __attr_always_inline___
-jerry_convert_eval_completion_to_retval (ecma_value_t completion, /**< completion of 'eval'-mode
-                                                                   *   code execution */
-                                         jerry_value_t *retval_p) /**< [out] api value */
-{
-  *retval_p = completion;
+#ifdef JMEM_STATS
+  if (jerry_init_flags & JERRY_INIT_MEM_STATS_SEPARATE)
+  {
+    jmem_stats_print ();
+    jmem_stats_reset_peak ();
+  }
+#endif /* JMEM_STATS */
 
-  return (ECMA_IS_VALUE_ERROR (completion)) ? JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION : JERRY_COMPLETION_CODE_OK;
-} /* jerry_convert_eval_completion_to_retval */
+  vm_init (bytecode_data_p);
 
-/**
- * @}
- */
+  return ecma_make_simple_value (ECMA_SIMPLE_VALUE_TRUE);
+} /* jerry_parse */
 
 /**
- * Copy string characters to specified buffer. It is the caller's responsibility
- * to make sure that the string fits in the buffer.
+ * Run Jerry in specified run context
  *
  * Note:
- *   '\0' could occur in characters.
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
  *
- * @return number of bytes, actually copied to the buffer.
+ * @return true - if run was successful
+ *         thrown error - otherwise
  */
-jerry_size_t
-jerry_string_to_char_buffer (const jerry_string_t *string_p, /**< string descriptor */
-                             jerry_char_t *buffer_p, /**< [out] output characters buffer */
-                             jerry_size_t buffer_size) /**< size of output buffer */
+jerry_value_t
+jerry_run (void)
 {
   jerry_assert_api_available ();
 
-  return ecma_string_copy_to_utf8_buffer (string_p, (lit_utf8_byte_t *) buffer_p, buffer_size);
-} /* jerry_string_to_char_buffer */
+  return vm_run_global ();
+} /* jerry_run */
 
 /**
- * Acquire string pointer for usage outside of the engine
- * from string retrieved in extension routine call from engine.
+ * Perform eval
  *
- * Warning:
- *         acquired pointer should be released with jerry_release_string
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
  *
- * @return pointer that may be used outside of the engine
+ * @return result of eval, may be error value.
  */
-jerry_string_t *
-jerry_acquire_string (jerry_string_t *string_p) /**< pointer passed to function */
+jerry_value_t
+jerry_eval (const jerry_char_t *source_p, /**< source code */
+            size_t source_size, /**< length of source code */
+            bool is_strict) /**< perform eval as it is called from strict mode code */
```

Comment:
I think the comment should be:
/**< source must conform with strict mode */
we call this from C, so strict mode is meaningless in that context